src/Diagrams/Segment.hs

```diff
@@ -580,3 +582,39 @@ instance (OrderedField n, Metric v)
 
             *: ()
 
+------------------------------------------------------------
+--  Serialize instances
+------------------------------------------------------------
+
+instance (Serialize (v n)) => Serialize (Segment Open v n) where
+  {-# INLINE put #-}
+  put segment = case segment of
+    Linear OffsetOpen    -> Serialize.put True
+    Cubic v w OffsetOpen -> Serialize.put False >> Serialize.put v >> Serialize.put w
+
+  {-# INLINE get #-}
+  get = do
+    isLinear <- Serialize.get
+    case isLinear of
+      True  -> return (Linear OffsetOpen)
+      False -> do
+        v <- Serialize.get
+        w <- Serialize.get
+        return (Cubic v w OffsetOpen)
```

Comment:
Would you prefer `case ... of True -> ... False -> ... ` or `if ... then ... else ...` here, and in the two other places where I am matching on a `Bool`?